File: BOJ/10816/A.py
import sys

sys.stdin = open('D:\Projects\AA\BOJ\\10816\input.txt', 'r')
input = sys.stdin.readline

N = int(input().rstrip())
d = {}
arr = list(map(int, input().rstrip().split()))
for i in range(len(arr)):
    num = arr[i]
    try:
        d[num] += 1
    except:
        d[num] = 1
M = int(input().rstrip())
r = list(map(int, input().strip().split()))
for i in range(len(r)):
    try:
        print(d[r[i]], end=' ')
    except:
        print(0, end=' ')

File: BOJ/10816/A.cpp
#include <iostream>
#include <map>
using namespace std;

int main()
{
    cin.tie(0);
    ios::sync_with_stdio(0);
    freopen("input.txt", "rt", stdin);
    int N, M;
    cin >> N;
    map<int, int> mp;
    for (int i = 0; i < N; i++)
    {
        int temp;
        cin >> temp;
        mp[temp]++;
    }
    cin >> M;
    for (int i = 0; i < M; i++)
    {
        int temp;
        cin >> temp;
        cout << mp[temp] << " ";
    }
}